### ISSUE TITLE
deferred error : (wrong-type-argument number-or-marker-p nil)

### ISSUE THREAD

**Issue by Voleking** [kaiwk/leetcode.el#32]
When I login in successfully, and run M-x Leetcode another time, I got this error: 
```
REQUEST [debug] REQUEST
REQUEST [debug] Run: curl --silent --include --location --compressed --cookie /Users/wangtianshu/.emacs.d/.cache/request/curl-cookie-jar --cookie-jar /Users/wangtianshu/.emacs.d/.cache/request/curl-cookie-jar --write-out \n(:num-redirects %{num_redirects} :url-effective "%{url_effective}") --form csrfmiddlewaretoken=@/var/folders/99/v69js2k510g5g724sf4f97500000gn/T/emacs-requestT0jfBH;filename= --form login=@/var/folders/99/v69js2k510g5g724sf4f97500000gn/T/emacs-requestkK1Trx;filename= --form password=@/var/folders/99/v69js2k510g5g724sf4f97500000gn/T/emacs-requestr3yxan;filename= --request POST --header User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:66.0) Gecko/20100101 Firefox/66.0 --header X-Requested-With: XMLHttpRequest --header Referer: https://leetcode.com/accounts/login --header X-CSRFToken: FpbaERbqkv9Ku9pw7ypJnO2JNumTSQ5ymdqsujHq7xJHOjWKR5tBFIAlA2xAmsMd https://leetcode.com/accounts/login
REQUEST [debug] REQUEST--CURL-CALLBACK event = finished

REQUEST [debug] REQUEST--CURL-CALLBACK proc = #<process request curl>
REQUEST [debug] REQUEST--CURL-CALLBACK buffer = #<buffer  *request curl*>
REQUEST [debug] REQUEST--CURL-CALLBACK symbol-status = nil
REQUEST [debug] REQUEST--CALLBACK
REQUEST [debug] (buffer-string) =
HTTP/2 200 
date: Wed, 04 Sep 2019 08:23:23 GMT
content-type: application/json
content-length: 17
vary: Cookie
x-frame-options: SAMEORIGIN
set-cookie: csrftoken=FpbaERbqkv9Ku9pw7ypJnO2JNumTSQ5ymdqsujHq7xJHOjWKR5tBFIAlA2xAmsMd; expires=Wed, 02 Sep 2020 08:23:23 GMT; Max-Age=31449600; Path=/; SameSite=Lax; Secure
strict-transport-security: max-age=31536000; includeSubDomains; preload
x-content-type-options: nosniff
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
server: cloudflare
cf-ray: 510e8fa50f258488-HKG

{"location": "/"}
REQUEST [debug] REQUEST-RESPONSE--CANCEL-TIMER
REQUEST [debug] -CLEAN-HEADER
REQUEST [debug] -CUT-HEADER
REQUEST [debug] error-thrown = nil
REQUEST [debug] -PARSE-DATA
REQUEST [debug] parser = buffer-string
REQUEST [debug] data = {"location": "/"}
REQUEST [debug] symbol-status = success
REQUEST [debug] Executing success callback.
REQUEST [debug] Executing complete callback.
REQUEST [debug] REQUEST
REQUEST [debug] Run: curl --silent --include --location --compressed --cookie /Users/wangtianshu/.emacs.d/.cache/request/curl-cookie-jar --cookie-jar /Users/wangtianshu/.emacs.d/.cache/request/curl-cookie-jar --write-out \n(:num-redirects %{num_redirects} :url-effective "%{url_effective}") --header User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:66.0) Gecko/20100101 Firefox/66.0 --header X-Requested-With: XMLHttpRequest --header Referer: https://leetcode.com/accounts/login https://leetcode.com/api/problems/all/
REQUEST [debug] REQUEST--CURL-CALLBACK event = finished

REQUEST [debug] REQUEST--CURL-CALLBACK proc = #<process request curl>
REQUEST [debug] REQUEST--CURL-CALLBACK buffer = #<buffer  *request curl*>
REQUEST [debug] REQUEST--CURL-CALLBACK symbol-status = nil
REQUEST [debug] REQUEST--CALLBACK
REQUEST [debug] (buffer-string) =
HTTP/1.0 200 Connection established

HTTP/2 200 
date: Wed, 04 Sep 2019 08:23:27 GMT
content-type: text/html; charset=utf-8
vary: Accept-Encoding
allow: GET, HEAD, OPTIONS
vary: Cookie
x-frame-options: SAMEORIGIN
set-cookie: csrftoken=FpbaERbqkv9Ku9pw7ypJnO2JNumTSQ5ymdqsujHq7xJHOjWKR5tBFIAlA2xAmsMd; expires=Wed, 02 Sep 2020 08:23:27 GMT; Max-Age=31449600; Path=/; SameSite=Lax; Secure
strict-transport-security: max-age=31536000; includeSubDomains; preload
x-content-type-options: nosniff
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
server: cloudflare
cf-ray: 510e8fb87aa3a2e2-HKG
content-encoding: gzip

{"user_name": "Voleking" ...}
REQUEST [debug] REQUEST-RESPONSE--CANCEL-TIMER
REQUEST [debug] -CLEAN-HEADER
REQUEST [debug] -CUT-HEADER
REQUEST [debug] error-thrown = (search-failed "^
")
REQUEST [debug] -PARSE-DATA
REQUEST [debug] parser = json-read
REQUEST [debug] data = nil
REQUEST [debug] symbol-status = error
REQUEST [debug] Executing error callback.
REQUEST [error] Error (error) while connecting to https://leetcode.com/api/problems/all/.
REQUEST [debug] Executing complete callback.
```

#### System Info :computer:
- OS: darwin
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 58d5bb920)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm

**Comment by kaiwk** [comment]
I can't reproduce it, no sure what happened : /

**Comment by Emp3404** [comment]
Hi @kaiwk . I have the same issue with the same debug output.

**Comment by kaiwk** [comment]
I have move on to [emacs-aio](https://github.com/skeeto/emacs-aio), so I think this problem would have been be solved : ) , see #33
